src/utils/parseError.ts

```diff
@@ -162,20 +163,41 @@ export const parseError = async (
 
   switch (e.code) {
     case EthersErrorType.CallExecption:
-      if (e.reason?.includes?.includes(EthersErrorMessage.ERC20Allowance)) {
-        return new TransactionError(
-          LifiErrorCode.AllowanceRequired,
+      const defaultErrorMessage = await getTransactionNotSentMessage(
+        step,
+        process
+      )
+      try {
+        if (!step?.action.fromChainId) {
+          throw new Error('Signer is not defined.')
+        }
+
+        const response = await fetchTxErrorDetails(
+          e.transactionHash,
+          step?.action.fromChainId
+        )
+
+        const errorMessage = response?.error_message ?? defaultErrorMessage
+
+        if (errorMessage?.includes(EthersErrorMessage.ERC20Allowance)) {
+          return new TransactionError(
+            LifiErrorCode.AllowanceRequired,
+            e.reason,
+            errorMessage,
+            e.stack
+          )
+        }
```

Comment:
I need one more clarification on this. 😄 Previously, we were checking if `e.reason` includes `EthersErrorMessage.ERC20Allowance` and now it looks like if tenderly returns an empty error message, then we will be checking `EthersErrorMessage.ERC20Allowance` against `defaultErrorMessage` returned by `getTransactionNotSentMessage` which will always be `false`. Am I missing something here?

Reply:
Resolved the issue. Thanks a lot for the careful review!